.github/workflows/release.yaml

```diff
@@ -32,3 +32,41 @@ jobs:
       - name: test-image
         run: |
           VERSION=3.6.99 ./scripts/test_images.sh
+      - name: save-image
+        run: |
+          docker image save -o /tmp/etcd-img.tar gcr.io/etcd-development/etcd
+      - name: upload-image
+        uses: actions/upload-artifact@v3
+        with:
+          name: etcd-img
+          path: /tmp/etcd-img.tar
+          retention-days: 1
+  trivy-scan:
+    needs: main
+    strategy:
+      fail-fast: false
+      matrix:
+        platforms: [amd64, arm64, ppc64le, s390x]
+    permissions:
+      security-events: write # for github/codeql-action/upload-sarif to upload SARIF results
+    runs-on: ubuntu-latest
+    steps:
+      - name: get-image
+        uses: actions/download-artifact@v3
+        with:
+          name: etcd-img
+          path: /tmp
+      - name: load-image
+        run: |
+          docker load < /tmp/etcd-img.tar
```

Comment:
It seems that the workflow does the same steps multiple times, but actually we can only do it once. Not sure how to resolve this. But overall not a big problem.

Reply:
I agree, I tried to club common steps in `main` job as much as possible but there is no direct data retention/sharing across jobs.